ActiveDirectory/Get-UserLogonSessionHistory.ps1

```diff
@@ -76,7 +76,7 @@ try
 					$_.TimeCreated -gt $LogonTime -and
 					$_.ID -in $SessionStopIds -and
 					(([xml]$_.ToXml()).Event.EventData.Data | where { $_.Name -eq 'TargetLogonId' }).'#text' -eq $LogonId
-				}) | select -First 1
+				}) | select -last 1
```

Comment:
Wouldn't this be the first event since that would be the event that would actually stop the session? For example, this may find user logoff events, computer shutdowns, etc. The first one to happen would be the one that actual ends the session.